src/extensions/registries/command-registry.ts

```diff
@@ -1,25 +1,25 @@
 // Extensions API -> Commands
 
-import type { Cluster } from "../../main/cluster";
-import type { Workspace } from "../../common/workspace-store";
 import { BaseRegistry } from "./base-registry";
-import { action } from "mobx";
+import { action, observable } from "mobx";
 import { LensExtension } from "../lens-extension";
+import { CatalogEntity } from "../../common/catalog-entity";
 
 export type CommandContext = {
-  cluster?: Cluster;
-  workspace?: Workspace;
+  entity?: CatalogEntity;
 };
 
 export interface CommandRegistration {
   id: string;
   title: string;
-  scope: "cluster" | "global";
+  scope: "entity" | "global";
   action: (context: CommandContext) => void;
   isActive?: (context: CommandContext) => boolean;
 }
 
 export class CommandRegistry extends BaseRegistry<CommandRegistration> {
+  @observable activeEntity: CatalogEntity;
```

Comment:
Why do we have this hear? It should be gotten on demand.

Reply:
This is pretty much unsolved problem within this PR (see todo list). We should have a way to pass an Entity that is currently active context. How to do that? (I start to feel we should solve this in a follow-up PR so that complexity of this PR does not grow too much)